added sandbox links to product sections

rahulchhabria opened the pull request:
@imatwawana and @Neo-Zhixing - can you give this a quick look, please? thank you.

rahulchhabria committed:
added sandbox links to product sections

imatwawana commented on src/docs/product/dashboards/index.mdx:
```diff
@@ -32,4 +32,4 @@ If you’d like to edit the default dashboard or build out multiple ones, each w
 ## Open Dashboard Widgets in Discover
 
-Each Dashboard widget has an ellipsis that opens a context menu. From here, you can open the widget in [Discover](/product/discover-queries/) for to get more information. If a widget contains more than one query, you'll be prompted to select which query to view in **Discover**.
+Each <SandboxLink scenario="dashboards" projectSlug="react">custom Dashboard</SandboxLink> widget has an ellipsis that opens a context menu. From here, you can open the widget in [Discover](/product/discover-queries/) for to get more information. If a widget contains more than one query, you'll be prompted to select which query to view in **Discover**.
```
Same comment as above.

```suggestion
Each Dashboard widget has an ellipsis that opens a context menu. From here, you can open the widget in [Discover](/product/discover-queries/) for to get more information. If a widget contains more than one query, you'll be prompted to select which query to view in **Discover**.
```

imatwawana commented on src/docs/product/performance/index.mdx:
```diff
@@ -10,23 +10,23 @@ redirect_from:
 </Note>
 
-The **Performance** page is the main view in [sentry.io](https://sentry.io) where you can search or browse for transaction data. The page displays graphs that visualize transactions or trends, as well as a table where you can view relevant transactions and drill down to more information about them.
+The **Performance** page is the main view in [sentry.io](https://sentry.io) where you can search or browse for <SandboxLink scenario="oneTransaction" projectSlug="react">transaction</SandboxLink> data. The page displays graphs that visualize transactions or trends, as well as a table where you can view relevant transactions and drill down to more information about them.
```
I would remove this link here. Generally speaking, if there's a link on a page about a different topic in docs, that link leads to those docs. So in cases where the Sandbox links are embedded in the topic of that page (e.g., in the word "performance" on the page about Performance), I don't think that breaks user expectations. But when it's embedded in the word "transactions" on this page when we have a section of content about transactions in Sentry Basics, it starts to break user expectations.

rahulchhabria replied:
hm... we've gotten a lot of feedback that customers aren't really familiar with transactions. i suggest leaving this here to help with any awareness. and the link is taking them to what a transaction looks like.

imatwawana replied:
Can we not link from actual transactions content?

imatwawana commented on src/docs/product/performance/web-vitals.mdx:
```diff
@@ -4,14 +4,14 @@ sidebar_order: 50
 
-[Web Vitals](https://web.dev/vitals/) are a set of metrics defined by Google to measure render time, response time, and layout shift. Each data point provides insights about the overall performance of your application.
+[Web Vitals](https://web.dev/vitals/) are a set of metrics defined by Google to measure render time, response time, and layout shift. Each data point provides insights about the overall <SandboxLink scenario="performance" projectSlug="react">performance</SandboxLink> of your application.
 
-The in-browser Sentry SDKs collect Web Vitals information (where supported) and add that information to frontend transactions. These vitals are then summarized in several graphs for a quick overview of how each frontend transaction is performing for your users.
+The in-browser Sentry SDKs collect <SandboxLink scenario="oneWebVitals" projectSlug="react">Web Vitals</SandboxLink> information (where supported) and add that information to frontend <SandboxLink scenario="oneTransaction" projectSlug="react">transactions</SandboxLink>. These vitals are then summarized in several graphs for a quick overview of how each frontend transaction is performing for your  ...
```
Same comment here as above. I'd strongly recommend against using the Sandbox link in the words "performance" and "transactions" here.

imatwawana commented on src/docs/product/releases/index.mdx:
```diff
@@ -8,9 +8,9 @@ redirect_from:
+A _release_ is a version of your code deployed to an environment. When you notify Sentry about a <SandboxLink scenario="oneRelease" projectSlug="react">release</SandboxLink>, you can easily identify new issues, regressions, whether an issue is resolved in the next release, and apply [source maps](/platforms/javascript/sourcemaps/).
 
-Releases also offer significant additional features when you fully [configure your SDK](/platform-redirect/?next=/configuration/releases/):
+<SandboxLink scenario="releases" projectSlug="react">Releases</SandboxLink> also offer significant additional features when you fully [configure your SDK](/platform-redirect/?next=/configuration/releases/):
```
The platform redirect link is breaking and I think it's because you're using HTML style tagging in the same line of code. Can the sandbox link be done in markdown instead? Please try committing this change and checking if the preview works after that. 

```suggestion
[Releases](https://try.sentry-demo.com/demo/start/?scenario=releases&projectSlug=react&source=docs) also offer significant additional features when you fully [configure your SDK](/platform-redirect/?next=/configuration/releases/):
```

Neo-Zhixing replied:
I wouldn't recommend using sandbox links directly as they may change. I was able to get this to work by doing this:
```
<SandboxLink scenario="releases" projectSlug="react">Releases</SandboxLink> also offer significant
 additional features when you fully <a href="/platform-redirect/?next=/configuration/releases/">configure your SDK</a>:
```

I will look into why markdown links didn't render.

imatwawana replied:
I'm pretty sure it's the same underlying behaviour that causes markdown to not render properly in a note if you don't add lines between the HTML tags of the note and the markdown. If the HTML tag is the first thing in the line, and there's no break, Gatsby doesn't seem to read the markdown.

imatwawana commented on src/docs/product/releases/index.mdx:
```diff
@@ -42,7 +42,7 @@ Each release links to one project. If a release has multiple projects, Sentry wi
 _Release health_ provides insight into the impact of crashes and bugs as it relates to your user's experience and reveals trends with each new issue. Monitor [Release Health](/product/releases/health/) by observing user adoption, usage of the application, percentage of crashes, and session data.
 
-You can view Release Health data either from the Issue Details page by selecting the commit ID listed under **Last Seen** or from the Releases homepage.
+You can view Release Health data either from the <SandboxLink scenario="oneIssue" projectSlug="react">Issue Details</SandboxLink> page by selecting the commit ID listed under **Last Seen** or from the Releases homepage.
```
Same comment here about linking to other topics. Strongly recommend the Sandbox link be removed here.

imatwawana reviewed (changes requested):
Great idea adding these links! I've made some comments about expected linking behaviour and some changes that should be made in that regard, in addition to one issue with a broken link.

rahulchhabria commented on src/docs/product/dashboards/index.mdx:
```diff
@@ -8,7 +8,7 @@ redirect_from:
 ---
 
-Dashboards provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
+<SandboxLink scenario="dashboards" projectSlug="react">custom Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```
hm... i can remove the custom bit, we only have one dashboard section in the product and the content is referencing https://try.sentry-demo.com/organizations/viable-elk/dashboards/ which is where i'm sending the link. so i think it is still relevant.

imatwawana replied:
Is the sandbox going to let them customize a dashboard?

rahulchhabria committed:
Update index.mdx

rahulchhabria committed:
Update index.mdx

removed the sandbox link till i can figure out how to get it to work with markdown.

imatwawana commented on src/docs/product/dashboards/index.mdx:
```diff
@@ -8,7 +8,7 @@ redirect_from:
 ---
 
-Dashboards provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
+<SandboxLink scenario="dashboards" projectSlug="react">custom Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```
```suggestion
[Dashboards](https://try.sentry-demo.com/demo/start/?scenario=dashboards&projectSlug=react&source=docs) provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```

imatwawana replied:
```suggestion
<SandboxLink scenario="dashboards" projectSlug="react">Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```

imatwawana committed:
Removing the word "custom"

imatwawana commented on src/docs/product/dashboards/index.mdx:
```diff
@@ -8,7 +8,7 @@ redirect_from:
 ---
 
-Dashboards provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
+<SandboxLink scenario="dashboards" projectSlug="react">Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```
@Neo-Zhixing do you have a solution for this one? This is another case where the HTML tags in the Sandbox link are breaking the link for the markdown. Ideally, we don't use html links between the docs themselves, rather just these short links.

Neo-Zhixing replied:
I think gatsby-mdx is interpreting `<SandboxLink>` as a block-level component making the links render incorrectly.
There are two workarounds:
- Don't use `<SandboxLink>` as the first thing in a paragraph. Put something else before it.
- Use `<a>` tags for links.

Obviously this is not ideal, so if it becomes a huge problem in the future, let me know and I'll go into the upstream packages to fix it.

imatwawana replied:
I've thrown a word in front of the link so hopefully that will fix this.

imatwawana replied:
Also @Neo-Zhixing, I'd venture to say this behaviour is already a problem (is a real problem in lists of links). Can I log it as an issue and assign it to you?

Neo-Zhixing replied:
Sure, put it on growth backlog and we will revisit this in the future.

imatwawana commented on src/docs/product/dashboards/index.mdx:
```diff
@@ -8,7 +8,7 @@ redirect_from:
 ---
 
-Dashboards provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
+<SandboxLink scenario="dashboards" projectSlug="react">Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```
```suggestion
Sentry's <SandboxLink scenario="dashboards" projectSlug="react">Dashboards</SandboxLink> provide you with a broad overview of your application’s health by allowing you to navigate through error and performance data across multiple projects. Dashboards are made up of one or more widgets, and each widget visualizes one or more [Discover queries](/product/discover-queries/).
```

imatwawana committed:
Added text ahead of Sandbox link to fix breaking behaviour
